src/lib/components/connect-button/connect-button.svelte

```diff
@@ -1,10 +1,31 @@
 <script lang="ts">
   import Button from 'radicle-design-system/Button.svelte';
-  import walletStore from '$lib/stores/wallet';
+  import wallet from '$lib/stores/wallet';
+  import ens from '$lib/stores/ens';
+
+  async function connect() {
+    await wallet.connect();
+  }
+
+  $: {
+    if ($wallet?.connected) {
+      const { provider, address } = $wallet ?? {};
+      if (!provider || !address) throw new Error('No provider after connection');
+
+      // Connect all stores to wallet
+      ens.connect(provider);
+
+      // Lookup own name
+      ens.lookup(address);
+    } else {
+      // Disconnect all stores from wallet
+      ens.disconnect();
+    }
+  }
```

Comment:
Not sure if this work of connecting & disconnecting stores that require a wallet here in the scope of the login button is necessarily the best place, as we may have other entrypoints for connecting and it might somehow happen that this reactive statement only starts listening after the store is initialized. Maybe moving this to a separate module somewhere might make more sense?

Reply:
>  it might somehow happen that this reactive statement only starts listening after the store is initialized

Not a concern after all as the statement will evaluate once on initialisation of the component, even without a change in any of the tracked variables.